ddtrace/tracer/sqlcomment.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+)
+
+// SQLCommentInjectionMode represents the mode of sql comment injection.
+type SQLCommentInjectionMode int
+
+const (
+	// SQLInjectionDisabled represents the comment injection mode where all injection is disabled.
+	SQLInjectionDisabled SQLCommentInjectionMode = 0
```

Comment:
Does `iota` not work here? Simply dropping the types and assignment below?

Reply:
It does but do we want users that might come here to consult what the numeric values are for setting their env variable to count the lines to know what the int value is? 

I also have a concern with accidental reordering that would essentially change the contract/API but I guess you'll always be there to catch those so maybe it's not a strong reason. 

Still, I wonder if there's a real benefit here to using iota.

Reply:
I think that's a good enough reason (checking the env. variable). Frankly though, I think it'd likely be a better experience to use more human values for the mode, for example: `off`/`disabled`, `service`, `full`. But this is also totally fine by me.

Reply:
That's a good point. I do prefer the string mode values as well but I can't remember why we went down the `int` route. Whatever it was, it likely doesn't apply anymore so I think I'll change it.